platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -1414,16 +1420,20 @@ private void serializeKeyToDirtyLeafIndex(
      * 		In case of trouble.
      */
     private void deserializeKeyToDirtyLeafIndex(
-            final Map<K, Mutation<VirtualLeafRecord<K, V>>> map, final SerializableDataInputStream in)
+            final Map<K, Mutation<K, VirtualLeafRecord<K, V>>> map,
+            final SerializableDataInputStream in,
+            final int version)
             throws IOException {
         final int sizeOfMap = in.readInt();
         for (int index = 0; index < sizeOfMap; index++) {
             final VirtualLeafRecord<K, V> leafRecord = in.readSerializable(false, VirtualLeafRecord::new);
-            leafRecord.setHash(in.readSerializable());
+            if (version == ClassVersion.ORIGINAL) {
+                // skip path
+                in.readLong();
```

Comment:
Should skip reading a hash, not a long

Reply:
No, in fact here we have to skip `long` because it's no longer a record and, therefore, it doesn't contain a path anymore. Here is how it used to be:
```
            final VirtualInternalRecord internalRecord = deleted ? null : new VirtualInternalRecord(path, hash);
            final Mutation<VirtualInternalRecord> mutation = new Mutation<>(null, key, internalRecord, mutationVersion);
```
Here is how it is now:
```
final Mutation<Long, Hash> mutation = new Mutation<>(null, key, hash, mutationVersion);
```